core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/DedupAggregator.java

```diff
@@ -87,6 +84,13 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     Schema outputSchema = getOutputSchema(inputSchema);
     FailureCollector collector = stageConfigurer.getFailureCollector();
     validateSchema(outputSchema, uniqueFields, functionInfo, collector);
+
+    if (functionInfo != null) {
```

Comment:
If input schema is null, then this check cannot be performed, since the function will require a field schema to start with, passing a null to it just doesn't make sense.